syntax = "proto3";

package Message; 

enum MsgDefine {
    S2C_MSG_BEGIN = 0;              //server message begin

    S2C_COMMON_ERROR = 50;          //错误码
    S2C_COMMON_NOTICE = 51;         //公告
    S2C_COMMON_BROADCAST = 52;      //全服广播

    S2C_USER_LOGIN= 100;            //登录验证返回
    S2C_USER_LOGOUT= 101;           //登录验证返回
    S2C_USER_HEARTBEAT = 102;       //心跳响应
    S2C_PLAYER_LOGIN = 103;         //通知角色登录
    S2C_PLAYER_INFO = 104;          //同步角色属性
    S2C_PLAYER_GETTHING = 105;      //奖励通知
    S2C_PLAYER_PREPAY = 106;        //微信预支付订单信息
    S2C_PLAYER_PREPAYALI = 107;     //支付宝预支付订单信息
    S2C_PLAYER_ACTIVITYDATA = 108;  //角色活动数据
    S2C_PLAYER_MONTHLYCARD = 109;   //角色月卡数据
    S2C_PLAYER_ACTIVITYPOINT = 110; //活跃度数据
    S2C_PLAYER_ACHIEVE = 111;       //角色成就

    S2C_PROP_LIST = 200;            //同步道具列表
    S2C_EQUIP_LIST = 201;           //同步已装备列表
    S2C_UNEQUIP_LIST = 202;         //同步未装备列表
    S2C_PROP_COMPOSE = 203;         //合成道具
    S2C_PROP_ENHANCE = 204;         //装备强化
    S2C_ENHANCE_INFO = 205;         //同步强化等级数据
    S2C_SHOPPING_INFO = 206;        //同步商品购买数据
    
    S2C_CHAT = 300;                 //聊天s2c
    S2C_FRIEND_LIST = 301;          //同步好友列表
    S2C_TASK_LIST = 302;            //同步任务列表
    S2C_SUITFRIEND_LIST = 303;      //同步推荐好友列表
    S2C_FRIENDAPPLY_LIST = 304;     //同步好友申请列表
    S2C_ADDFRIEND_SUC = 305;        //发送添加好友成功
    S2C_FRIEND_STATUS = 306;        //同步好友上下线状态
    S2C_TASK_SYNCSTORY = 307;       //同步剧情
    S2C_FRIEND_FLOWER = 308;        //同步好友送花

    S2C_SKILL_LIST = 320;           //同步主动技能列表
    S2C_BASE_GONG = 321;            //同步基础功法信息
    S2C_HEART_GONG = 322;           //同步心法信息
    S2C_GUILD_SKILL = 323;          //同步帮派技能

    S2C_FIGHT_UNITLIST = 330;       //战斗单位列表
    S2C_FIGHT_PROCESS = 331;        //战斗过程
    S2C_FIGHT_RESULT = 332;         //战斗结果
    S2C_FIGHT_AUTO = 333;           //自动战斗
    S2C_FIGHT_PK = 334;             //切磋请求
    S2C_FIGHT_REJECTPK = 335;       //拒绝切磋请求

    S2C_TEAM_INFO = 350;            //同步队伍信息
    S2C_TEAM_APPLY = 351;           //转发申请入队信息
    S2C_TEAM_INVITE = 352;          //转发送邀请信息
    S2C_TEAM_LEAVE = 353;           //向客户端同步离队类型
    S2C_TEAM_GIVE = 354;            //同步转交队长
    S2C_TEAM_MOVE = 355;            //同步队长移动
    S2C_TEAM_MEMBERSTATUS = 356;    //同步队员上下线状态

    S2C_MAIL_LIST = 400;            //同步邮件列表

    S2C_SCENE_MOVE = 500;           //移动请求返回
    S2C_SCENE_NPC = 501;            //同步场景中npc数据
    S2C_SCENE_PLAYER = 502;         //同步场景中player数据
    S2C_SCENE_EVENT = 503;          //同步场景事件
    S2C_SCENE_ENTER = 504;          //通知主角进入场景

    S2C_PALACE_INFO = 600;          //同步副本信息
    S2C_PALACE_PLAYERINFO = 601;    //同步角色副本信息
    S2C_TRAINROOM_INFO = 610;       //同步角色练功房数据

    S2C_PET_INFO = 701;             //同步唯一宠物信息
    S2C_PET_LIST = 702;             //同步宠物列表

    S2C_ARENA_PLAYERINFO = 800;     //同步玩家竞技场数据
    S2C_ARENA_TARGETLIST = 801;     //同步玩家竞技场挑战列表
    S2C_ARENA_TOPLIST = 802;        //同步竞技场前10列表
    S2C_ARENA_FIGHTRESULT = 803;    //同步竞技场战斗

    S2C_HUASHAN_TEAMINFO = 900;     //同步华山论剑队伍数据
    S2C_HUASHAN_TOPLIST = 901;      //同步华山论剑排行数据

    S2C_TOWER_INFO = 1000;          //爬塔信息

    S2C_RANKLIST_INFO = 1001;       //同步排行数据

    S2C_GUILD_SELFINFO = 1050;      //玩家帮会信息
    S2C_GUILD_INFO = 1051;          //本帮会信息
    S2C_GUILD_MEMBERLIST = 1052;    //本帮成员列表
    S2C_GUILD_OTHERLIST = 1053;     //其他帮会列表
    S2C_GUILD_APPLYERLIST = 1054;   //本帮申请列表
    S2C_GUILD_MANAGERLIST = 1055;   //本帮管理列表
    S2C_GUILD_OPERATION = 1056;     //同步单一操作
    S2C_GUILD_RECORD = 1057;        //同步帮派记录
    S2C_GUILD_SINGLE = 1058;        //同步查找帮派
    S2C_GUILD_WAROPPOTEAM = 1059;   //同步帮战对方队伍信息
    S2C_GUILD_WARTEAMCHANGE = 1060; //同步帮战对方队伍变化
    S2C_GUILDWAR_SCORELIST = 1061;  //同步帮会积分表
    S2C_GUILDWAR_INFO = 1062;       //同步帮战状态时间信息
    S2C_GUILDWAR_RET = 1063;        //同步帮战结果和时时分
    S2C_GUILDWAR_TEAMINFO = 1064;   //队伍信息及本帮时时分
    S2C_GUILD_WARSELFTEAM = 1065;   //同步本帮队伍信息
    S2C_GUILD_WARSELFCHANGE = 1066; //同步帮战对方队伍变化
    S2C_GUILDMAZE_INFO = 1067;      //同步帮派迷阵信息
    S2C_GUILDMAZE_EVENT = 1068;     //帮派迷阵触发事件

    S2C_HUNT_INFO = 1100;           //同步全局狩猎数据
    S2C_HUNT_PLAYERINFO = 1101;     //同步玩家狩猎数据

    S2C_ZJZD_TOPLIST = 1200;        //紫禁之巅同步排行数据
    S2C_ZJZD_PLAYERINFO = 1201;     //紫禁之巅同步个人数据

    S2C_FISH_GET = 1250;            //通知钓鱼id
    S2C_FISH_INFO = 1251;           //同步玩家钓鱼数据

    S2C_WORLDBOSS_INFO = 1300;      //世界boss数据
    S2C_WORLDBOSS_TOTALRANK = 1301; //世界boss总伤害排行
    S2C_WORLDBOSS_SINGLERANK = 1302;//世界boss单场伤害排行

    S2C_BATHE_INFO = 1350;          //同步温泉数据
    S2C_BATHE_ALLINFO = 1351;       //同步全部温泉数据

    S2C_SIGN_INFO = 1400;           //同步签到数据
    S2C_TITLE_INFO = 1401;          //同步称号数据
    S2C_OFFICER_INFO = 1402;        //同步官职信息
    S2C_QUESTION_INFO = 1430;       //同步答题数据
    S2C_NATIONWAR_INFO = 1440;      //同步阵营战信息
	S2C_NATIONWAR_TEAMSTATUS = 1441;//同步阵营战队伍信息
    S2C_NATIONWAR_TEAMTOP = 1442;   //同步阵营前五队伍

    S2C_RUNE_PLAYER = 1500;         //同步主角穿戴的符文
    S2C_RUNE_PET = 1501;            //同步门客穿戴的符文
    S2C_RUNE_LIST = 1502;           //同步未穿戴的符文

    S2C_BODILY_QIINFO = 1550;       //同步冥想信息
    S2C_BODILY_INFO = 1551;         //同步经脉信息
    S2C_BODILY_UPGRADERET = 1552;   //同步经脉升级结果

    S2C_RAIDER_TEAMINFO = 1600;     //夺宝奇兵同步队伍数据
    S2C_RAIDER_GAIN = 1601;         //夺宝奇兵同步获得资源事件
    S2C_RAIDER_STEAL = 1602;        //夺宝奇兵同步偷盗事件
    S2C_RAIDER_TRAPSET = 1603;      //夺宝奇兵同步放置陷阱事件
    S2C_RAIDER_TRAPTRIGGER = 1604;  //夺宝奇兵同步踩中陷阱事件
    S2C_RAIDER_PLUNDER = 1605;      //夺宝奇兵同步掠夺事件

    S2C_MSG_END = 9999;             //server message end
    C2S_MSG_BEGIN = 10000;          //client message begin

    //user & player
    C2S_GUEST_LOGIN = 10001;        //请求游客登录
    C2S_USER_LOGIN = 10002;         //请求账号登录
    C2S_USER_LOGOUT = 10003;        //账号登出
    C2S_USER_HEARTBEAT = 10004;     //心跳请求
    C2S_PLAYER_CREATE = 10005;      //创建角色
    C2S_PLAYER_LOGIN = 10006;       //角色登录 
    C2S_PLAYER_LOGOUT = 10007;      //角色登出
    C2S_PLAYER_GUIDE = 10008;       //同步引导步骤
    C2S_PLAYER_GIFTCODE = 10009;    //使用礼包码
    C2S_PLAYER_RECHARGE = 10010;    //充值
    C2S_PLAYER_RENAME = 10011;      //改名
    C2S_PLAYER_ACTIVITYPRIZE = 10012; //请求领取活跃度奖励
    C2S_PLAYER_ACHIEVEPRIZE = 10013;//领取成就奖励
    C2S_PLAYER_ACTIVITYJOIN = 10014;//参加运营活动

    C2S_NPC_TRIGGER = 10050;        //触发npc

    //prop
    C2S_PROP_USE = 10100;           //使用道具
    C2S_PROP_BUY = 10101;           //购买道具
    C2S_PROP_SELL = 10102;          //出售道具
    C2S_PROP_COMPOSE = 10103;       //合成道具
    C2S_PROP_DECOMPOSE = 10104;     //分解道具
    C2S_PROP_EQUIP = 10105;         //穿装备
    C2S_PROP_UNEQUIP = 10106;       //脱装备
    C2S_PROP_SETGEM = 10107;        //镶嵌宝石
    C2S_PROP_UNSETGEM = 10108;      //拆除宝石
    C2S_PROP_COMPOSEGEM = 10109;    //合成包裹中的宝石
    C2S_PROP_COMPOSEEQUIPGEM = 10110;//合成镶嵌中的宝石
    C2S_PROP_ENHANCE = 10111;       //强化装备
    C2S_PROP_ENHANCEBLESS = 10112;  //祝福强化

    //mail
    C2S_MAIL_SEND = 10200;          //发送邮件
    C2S_MAIL_READ = 10201;          //读取邮件
    C2S_MAIL_DELETE = 10202;        //删除邮件
    C2S_MAIL_EXTRACT = 10203;       //提取道具
    //char & friend
    C2S_CHAT = 10300;               //聊天c2s
    C2S_FRIEND_ADD = 10301;         //添加好友
    C2S_FRIEND_DELETE = 10302;      //删除好友
    C2S_FRIEND_RESULT = 10303;      //申请回执
    C2S_FRIEND_REQLIST = 10304;     //请求好友列表信息
    C2S_FRIEND_SENDFLOWER = 10305;  //给好友送花

    //task
    C2S_TASK_ACCEPT = 10310;        //接取任务
    C2S_TASK_FINISH = 10311;        //完成任务
    C2S_TASK_CANCEL = 10312;        //取消任务
    C2S_TASK_FIGHT = 10313;         //触发任务战斗
    C2S_TASK_SYNCSTORY = 10314;     //同步任务剧情
    
    //fight
    C2S_FIGHT_PK = 10328;           //发起切磋
    C2S_FIGHT_PKRET = 10329;        //切磋回执
    C2S_FIGHT_USESKILL = 10330;     //使用技能
    C2S_FIGHT_ROUNDREADY = 10331;   //播报完毕
    C2S_SKILL_UPGRADE = 10332;      //主动技能升级
    C2S_BASEGONG_ADDPOINT = 10333;  //基础功法加点
    C2S_HEARTGONG_RESET = 10334;    //重置基础功法加点
    C2S_HEARTGONG_UPGRADE = 10335;  //心法升级
    C2S_FIGHT_AUTO = 10336;         //同步自动战斗
    C2S_FIGHT_CALLPET = 10337;      //战斗中召唤宠物
    C2S_GUILDSKILL_UPGRADE = 10338; //升级帮会技能
    C2S_FIGHT_GIVEUP = 10339;       //战斗认输

    //team
    C2S_TEAM_CREATE = 10340;        //创建队伍
    C2S_TEAM_APPLY = 10341;         //申请入队
    C2S_TEAM_APPLYRESULT = 10342;   //处理申请
    C2S_TEAM_INVITE = 10343;        //邀请入队
    C2S_TEAM_INVITERESULT = 10344;  //处理邀请
    C2S_TEAM_QUIT = 10345;          //退出队伍
    C2S_TEAM_KICK = 10346;          //踢出队伍
    C2S_TEAM_DISSOLVE = 10347;      //解散队伍
    C2S_TEAM_GIVE = 10348;          //转交给新队长
    C2S_TEAM_GIVERESULT = 10349;    //转交给新队长回执
    C2S_TEAM_REPLACE = 10350;       //队员申请成为队长
    C2S_TEAM_FORMATION = 10351;     //队长改变阵型
    C2S_TEAM_PLAYERMATCH = 10352;   //单人匹配
    C2S_TEAM_TEAMMATCH = 10353;     //队伍招募

    C2S_SCENE_MOVE = 10400;         //请求移动
    C2S_SCENE_SWITCH = 10401;       //请求传送场景
    C2S_PALACE_ENTER = 10402;       //请求进入副本
    C2S_PALACE_QUIT = 10403;        //请求退出副本
    C2S_PALACE_FINISH = 10404;      //请求完成副本关卡
    C2S_TRAINROOM_FIGHT = 10410;    //请求练功房战斗
    C2S_TRAINROOM_ENTER = 10411;    //请求进入练功房

    C2S_PET_UPGRADEQUA = 10480;     //提升宠物品质
    C2S_PET_SETFIGHT = 10481;       //设置出战宠物
    C2S_PET_EXCHANGE =  10482;      //宠物兑换
    C2S_PET_UPGRADESTAR = 10483;    //宠物升星
    C2S_PET_USEEXPPROP = 10484;     //使用宠物经验道具

    C2S_ARENA_ENTER = 10500;        //请求进入竞技场
    C2S_ARENA_FIGHT = 10501;        //请求竞技场挑战对手
    C2S_ARENA_TOPLIST = 10502;      //请求竞技场前20数据
    C2S_ARENA_RANKPRIZE = 10503;    //请求竞技场领取排名奖励

    C2S_HUASHAN_ENTER = 10600;      //请求进入华山论剑地图
    C2S_HUASHAN_QUIT = 10601;       //请求退出华山论剑地图
    C2S_HUASHAN_TOPLIST = 10602;    //请求华山论剑排行数据
    C2S_HUASHAN_MATCH = 10603;      //请求华山论剑参与匹配

    C2S_TOWER_FIGHT = 10650;        //爬塔触发战斗
    C2S_TOWER_RESET = 10651;        //重置爬塔

    C2S_RANKLIST_INFO = 10700;      //请求排行数据

    C2S_GUILD_CREATE = 10750;       //创建帮会
    C2S_GUILD_APPLY = 10751;        //申请加入帮会
    C2S_GUILD_APPLYRET = 10752;     //处理申请
    C2S_GUILD_SETJOB = 10753;       //官员任命
    C2S_GUILD_SETNOTICE = 10754;    //设置公告
    C2S_GUILD_REQOTHER = 10755;     //请求其他公会列表(前15默认发送)
    C2S_GUILD_QUIT = 10756;         //退出帮会
    C2S_GUILD_DISSOLVE = 10757;     //解散帮会
    C2S_GUILD_RENAME = 10758;       //帮会重命名
    C2S_GUILD_KICK = 10759;         //踢出帮会
    C2S_GUILD_REQMEMBER = 10760;    //请求帮会成员列表
    C2S_GUILD_CLEARAPPLY = 10761;   //清空申请列表
    C2S_GUILD_CHANGELEADER = 10762; //禅让帮主
    C2S_GUILD_FINDGUILD = 10763;    //ID查找帮会
    C2S_GUILDWAR_ENTERWARMAP = 10764;   //申请进入帮战地图
    C2S_GUILDWAR_REQOPPOINFO = 10765;   //请求对手信息
    C2S_GUILDWAR_REQFIGHT = 10766;      //请求帮会战斗
    C2S_GUILDWAR_REQSCORESORT = 10767;  //请求帮会积分列表
    C2S_GUILD_GETBOX = 10768;           //领取繁荣度宝箱
	C2S_GUILD_REPLACELEADER = 10769;	//弹劾帮主
    C2S_GUILD_MAZESELECT = 10770;       //帮派迷阵选择

    C2S_HUNT_FIGHT = 10800;         //狩猎请求战斗

    C2S_FISH_GET = 10850;           //客户端开始钓鱼
    C2S_FISH_GO = 10851;            //客户端完成钓鱼
    C2S_FISH_INFO = 10852;          //客户端请求图鉴
    C2S_FISH_JOIN = 10853;          //客户端请求进入渔场

    C2S_ZJZD_ENTER = 10900;         //紫禁之巅请求进入
    C2S_ZJZD_MATCH = 10901;         //紫禁之巅请求匹配
    C2S_ZJZD_TOPLIST = 10902;       //紫禁之巅请求排行数据

    C2S_WORLDBOSS_ENTER = 10950;    //世界boss进入地图
    C2S_WORLDBOSS_FIGHT = 10951;    //世界boss进入战斗
    C2S_WORLDBOSS_TOTALRANK = 10952;//世界boss总伤害排行
    C2S_WORLDBOSS_SINGLERANK = 10953; //世界boss单场伤害排行
    C2S_WORLDBOSS_INFO = 10954;     //世界boss血量

    C2S_BATHE_OCCUPY = 11000;       //温泉占领位置
    C2S_BATHE_LEAVE = 11001;        //温泉离开位置
    C2S_BATHE_INFO = 11002;         //温泉请求数据
    C2S_BATHE_ALLINFO = 11003;      //温泉请求全部数据

    C2S_REQ_SIGN = 11050;           //请求签到
    C2S_REQ_REMEDYSIGN = 11051;     //请求补签
    C2S_TITLE_SET = 11052;          //设置当前称号
    C2S_TIMEOUT_REQ = 11053;        //到期请求
    C2S_OFFICER_REWARD = 11054;     //领取本周官职奖励
    C2S_QUESTION_REQ = 11055;       //发送答案

    C2S_NATION_SELECT = 11060;      //选择阵营
    C2S_NATION_ENTERMAP = 11061;    //申请进入阵营战地图
    C2S_NATION_MATCHFIGHT = 11062;  //匹配阵营战
    C2S_NATION_REQWARINFO = 11063;  //请求阵营战信息
    C2S_NATION_REQTOPTEAM = 11064;  //请求阵营战队伍榜TOP5

    C2S_RUNE_EQUIPPLAYER = 11100;   //请求主角穿戴符文
    C2S_RUNE_EQUIPPET = 11101;      //请求门客穿戴符文
    C2S_RUNE_DECOMPOSE = 11102;     //请求分解符文
    C2S_RUNE_RESET = 11103;         //请求洗练符文
    C2S_RUNE_SMELTING = 11104;      //请求熔炼符文
    C2S_RUNE_UPLEVEL = 11105;       //请求符文升级
    C2S_RUNE_UPQUALITY = 11106;     //请求符文升品
    C2S_RUNE_UPSTARTS = 11107;      //请求符文升星

    C2S_BODILY_GETQI = 11150;       //请求获取真气
    C2S_BODILY_UPGRADE = 11151;     //请求经脉升级
    C2S_BODILY_QISPEEDUP = 11152;   //请求真气加速度

    C2S_RAIDER_JOIN = 11200;        //夺宝奇兵请求参加
    C2S_RAIDER_USETRAP = 11201;     //夺宝奇兵请求使用陷阱道具
    C2S_RAIDER_USESTEAL = 11202;    //夺宝奇兵请求使用偷偷令
    C2S_RAIDER_USEPLUNDER = 11203;  //夺宝奇兵请求使用掠夺令
    
    C2S_MSG_END = 30000;            //client message end
}
